### PR TITLE
Minor VTables Enhancements

### PR DESCRIPTION
**Pull request by thestr4ng3r**
A bit of restructuring (reducing pyramids of doom)
Remove fixed restriction to x86+elf
Add more possible sections that may contain vtables
#6851

### CODE REVIEW

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -6416,29 +6416,25 @@ static void r_anal_virtual_functions(void *core, const char* input) {
...
+	case '*'://av*
+		r_core_anal_list_vtables_all (core);
+		break;
+	case 'j': //avj
+		r_core_anal_list_vtables (core, true);
+		break;
+	case 'r': //avr
+		r_core_anal_print_rtti (core);
+		break;
+	case '\0': //av
```
Comment (xarkes) [inline libr/core/cmd_anal.c]: I think writing `// "av"` is better for grepping.

**Comment by radare** [comment]
Agree

**Comment by thestr4ng3r** [comment]
already updated

File: libr/core/anal_vt.c [hunk 2]
```diff
@@ -52,28 +63,31 @@ static int isVtableStart(RCore *core, ut64 curAddress) {
...
+		return false;
+	}
+	r_list_foreach (xrefs, xrefIter, xref) {
+		// section in which currenct xref lies
+		if (inTextSection (core, xref->addr)) {
+			r_io_read_at (core->io, xref->addr, buf, VTABLE_BUFF_SIZE);
+			if (!r_asm_disassemble (core->assembler, &asmop, buf, VTABLE_BUFF_SIZE) > 0) {
+				continue;
+			}
+			if ((!strncmp (asmop.buf_asm, "mov", 3)) ||
```
Comment (XVilka) [inline libr/core/anal_vt.c]: do this only for x86, also I would use AnalOp instead.

Comment (thestr4ng3r) [reply libr/core/anal_vt.c]: Exactly, that would be my next step.

**Comment by XVilka** [comment]
@thestr4ng3r also it may be a better idea to move this code into `libr/anal` instead of `libr/core`.

**Pull request merged**